ExampleOutput/Commands.g.cs

```diff
@@ -2,30 +2,35 @@
 using System.CommandLine;
 using System.CommandLine.Invocation;
 using System.CommandLine.Parsing;
-using DemoHandlersUpdated;
+using Jackfruit;
 
-namespace Jackfruit
+namespace DemoHandlersUpdated
 {
-    public partial class FranshiseCommand : GeneratedCommandBase<FranshiseCommand.Result>
+    public partial class CliRoot : GeneratedCommandBase<CliRoot, CliRoot.Result>
     {
-        private FranshiseCommand() : base("rootCommand")
+        public static CliRoot Create(Delegate methodToRun)
         {
+            return Create();
         }
 
-        public static FranshiseCommand Create()
+        private CliRoot() : base("rootCommand")
         {
-            var command = new FranshiseCommand();
+        }
+
+        public static CliRoot Create()
+        {
+            var command = new CliRoot();
             command.GreetingArgument = new Argument<string>("greetingArg");
```

Comment:
Note that greeting and the validation for greeting exists once in the root and is available in all subcommands.